src/main/kotlin/io/sc3/peripherals/posters/printer/PosterPrinterBlockEntity.kt

```diff
@@ -0,0 +1,336 @@
+package io.sc3.peripherals.posters.printer
+
+import dan200.computercraft.api.peripheral.IComputerAccess
+import io.sc3.library.ext.optCompound
+import io.sc3.library.networking.NetworkUtil.sendToAllTracking
+import io.sc3.peripherals.Registration.ModBlockEntities.posterPrinter
+import io.sc3.peripherals.Registration.ModItems
+import io.sc3.peripherals.config.ScPeripheralsConfig.config
+import io.sc3.peripherals.posters.PosterItem
+import io.sc3.peripherals.posters.PosterItem.Companion.POSTER_KEY
+import io.sc3.peripherals.posters.PosterPrintData
+import io.sc3.peripherals.util.BaseBlockEntity
+import io.sc3.peripherals.util.ImplementedInventory
+import net.minecraft.block.Block
+import net.minecraft.block.BlockState
+import net.minecraft.entity.player.PlayerEntity
+import net.minecraft.entity.player.PlayerInventory
+import net.minecraft.inventory.Inventories
+import net.minecraft.inventory.SidedInventory
+import net.minecraft.item.ItemStack
+import net.minecraft.item.Items
+import net.minecraft.nbt.NbtCompound
+import net.minecraft.network.Packet
+import net.minecraft.network.listener.ClientPlayPacketListener
+import net.minecraft.network.packet.s2c.play.BlockEntityUpdateS2CPacket
+import net.minecraft.screen.NamedScreenHandlerFactory
+import net.minecraft.screen.PropertyDelegate
+import net.minecraft.screen.ScreenHandler
+import net.minecraft.server.world.ServerChunkManager
+import net.minecraft.text.Text
+import net.minecraft.util.ItemScatterer
+import net.minecraft.util.collection.DefaultedList
+import net.minecraft.util.math.BlockPos
+import net.minecraft.util.math.Direction
+import net.minecraft.world.World
+import org.slf4j.LoggerFactory
+import java.util.*
+import java.util.concurrent.ConcurrentHashMap
+
+class PosterPrinterBlockEntity(
+  pos: BlockPos,
+  state: BlockState
+) : BaseBlockEntity(posterPrinter, pos, state), NamedScreenHandlerFactory, ImplementedInventory, SidedInventory {
+  private val inventory = DefaultedList.ofSize(INV_SIZE, ItemStack.EMPTY)
+
+  val log = LoggerFactory.getLogger("ScPeripherals/PosterPrinterBlockEntity")!!
+
+  /** Set of computers that are attached as a peripheral to the printer, so they may receive print state events. */
+  val computers: MutableSet<IComputerAccess> = Collections.newSetFromMap(ConcurrentHashMap())
+
+  var animatingPosterId: String? = null
+  var animationStartTime: Long = 0
+
+  var data: PosterPrintData = PosterPrintData(
+    null, null, ByteArray(128*128) { 0 }
```

Comment:
I could be wrong, but isn't it already initialized with zeroes? I'd imagine it would be faster to leave `{ 0 }` out

```suggestion
    null, null, ByteArray(128*128)
```